spec/controllers/admin/settings_controller_spec.rb

```diff
@@ -0,0 +1,181 @@
+      it "denies random admin access" do
+        admin.update(roles: [])
+        fake_login_admin(admin)
+        get :index
+        it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+      end
+    end
+
+    context 'when admin has correct authorization' do
+      it "allows admins to access index" do
+        admin.update(roles: ['policy_and_abuse'])
+        fake_login_admin(admin)
+        get :index
+        expect(response).to have_http_status(:success)
+      end
+    end
+  end
+
+  describe "PUT #update" do
+    let(:admin) { create(:admin) }
+    let(:setting) { create(:admin_setting, last_updated_by: admin.id) }
+
+    context 'when admin does not have correct authorization' do
+      it "denies random admin access" do
+        admin.update(roles: [])
+        fake_login_admin(admin)
+        put :update, params: { id: setting.id, admin_setting: { last_updated_by: admin.login } }
+
+        it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+      end
+    end
+
+    context 'when admin has correct authorization' do
+      context 'when admin has superadmin role' do
+        it "allows superadmins to update all settings" do
+          admin.update(roles: ['superadmin'])
+          fake_login_admin(admin)
+
+          put :update, params: {
+            id: setting.id,
+            admin_setting: {
+              account_creation_enabled: "1",
+              creation_requires_invite: "1",
+              request_invite_enabled: "0",
+              invite_from_queue_enabled: "1",
+              invite_from_queue_number: "10",
+              invite_from_queue_frequency: "7",
+              days_to_purge_unactivated: "7",
+              disable_support_form: "1",
+              disabled_support_form_text: "Disable support",
+              suspend_filter_counts: "0",
+              tag_wrangling_off: "0",
+              downloads_enabled: "1",
+              enable_test_caching: "0",
+              cache_expiration: "10",
+              hide_spam: "1"
+            } 
+          }
+
+          it_redirects_to_with_notice(admin_settings_path, "Archive settings were successfully updated.")
+        end
+      end
+
+      context 'when admin has policy_and_abuse role' do
+        it "prohibits admins with policy_and_abuse role to update forbidden settings" do
+          admin.update(roles: ['policy_and_abuse'])
+          fake_login_admin(admin)
+
+          put :update, params: {
+            id: setting.id,
+            admin_setting: {
+              suspend_filter_counts: "0",
+              tag_wrangling_off: "0",
+              downloads_enabled: "1",
+              enable_test_caching: "0",
+              cache_expiration: "10",
+              hide_spam: "1"
+            } 
+          }
+
+          expect(response).to render_template('index')
```

Comment:
Just for the heck of it, as long as this file needs some other changes, can you do `render_template(:index)` instead?